public/index.html

```diff
@@ -29,6 +29,7 @@
   <body>
     <noscript>You need to enable JavaScript to run this app.</noscript>
     <div id="root"></div>
+    <div id="portal"></div>
```

Comment:
I won't say this is "wrong" but I personally are not a fan of declaring this outside of `root`. The react app is based on `root` and what you're doing here is rendering some react components outside of the scope which doesn't make sense. 

There are two ways to approach making a modal. One is to make it a "global" state and have some kind of "modalManager" to decide what modal to render. Another way is to have the modal being rendered locally in each component that needs it. Either way you can use css to achieve the overlay effect you want for modal (tip: `z-index`). 

Seems like what you're trying to do here is the first approach (global state). What I would do is create a `context` in your `App` component such as `modalContext` and use it to control if a modal should be rendered or not. This will be a good time to understand how context work and when to use them.